integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/InputFieldsCollector.java

```diff
@@ -104,4 +112,25 @@ private static List<DatasetIdentifier> extractDatasetIdentifier(CatalogTable cat
               catalogTable.location().getPath(), PlanUtils.namespaceUri(catalogTable.location())));
     }
   }
+
+  /* Similar to the InsertIntoHadoopFsRelationVisitor and LogicalRelationDatasetBuilder
+   * We need to handle a HadoopFsRelation by extracting that paths it traverses
+   * to identify the datasets being used.
+   */
+  private static List<DatasetIdentifier> extractDatasetIdentifier(HadoopFsRelation relation) {
+    List<DatasetIdentifier> inputDatasets = new ArrayList<DatasetIdentifier>();
+    List<Path> paths =
+        JavaConversions.asJavaCollection(relation.location().rootPaths()).stream()
+            .collect(Collectors.toList());
+
+    for (Path p : paths) {
+      String namespace = PlanUtils.namespaceUri(p.toUri());
+      inputDatasets.add(new DatasetIdentifier(p.toUri().getPath(), namespace));
+    }
+    if (inputDatasets.isEmpty()) {
```

Comment:
Shouldn't `return inputDatasets;` be just enough?

Reply:
🤦yeah... that's a good one.  Thank you for catching and improving, Pawel.